frontend/components/shared/MRForm.vue

```diff
@@ -1,60 +1,74 @@
 <script lang="ts" setup>
 import { ref, computed } from "vue";
 import { BSButton } from "cdh-vue-lib";
-import type { FormConfig, FormStep } from "../form/types";
 import FormStepper, { type FormStepperConfig } from "./FormStepper.vue";
+import { type GetFormQuery } from "~/generated/gql/graphql";
+
+type QueriedForm = NonNullable<GetFormQuery["form"]>;
+type ParentStep = NonNullable<QueriedForm["steps"][number]>;
+type Substep = NonNullable<ParentStep["substeps"][number]>;
+
+export type Step = ParentStep | Substep;
 
 interface Props {
-    formConfig: FormConfig;
+    form: QueriedForm;
 }
 
 const props = defineProps<Props>();
 
+const steps = computed<ParentStep[]>(() =>
+    props.form.steps.map((step) => ({
+        ...step,
+        substeps: step.substeps.map((substep) => ({
+            ...substep,
+        })),
+    })),
+);
```

Comment:
I like it when JS gets all functional like this. Though a recursive implementation would be more generic once we increase step depth > 1.

Though I can see why this is better for now. No need to worry about levels to compute the indentation and whatnot.